### ISSUE TITLE
Annouces IP addresses that don't match interfaces advertised on

### ISSUE THREAD

**Issue by karlp** [probonopd/announce#11]
If you have two interfaces, say, 192.168.49.* and 192.168.255.*, announce is picking the first non-loopback address and advertising that as the ip.  It should be using the appropriate IP address for the interface it's announcing on, not just any old ip.

**Comment by probonopd** [comment]
Thanks for catching this; do you have an idea on how to implement this?

**Comment by karlp** [comment]
no, just something quick I came across while evaluating alternatives to avahi-daemon.   Currently working on the openwrt mdnsd tests, which, as noted are undocumented, but I think might actually work.

**Comment by probonopd** [comment]
A fuller implementation is avahi, I am sure you have considered that. This here is the bare minimum if you want to squeeze every KB.

**Comment by karlp** [comment]
No, I've been using avahi-daemon, but I want to save space.  Unfortunately, I can't really say this is the "bare minimum" if it's advertising the wrong IPs.  I suspect you need to have a mdns instance per interface, not just a single instance?  Anyway.
